src/loader.c

```diff
@@ -969,11 +969,9 @@ int get_cgroup_data(const char *pid_cgroup, char *pod_uid, char *container_id,
     buffer[strlen(buffer) - 1] = '\0';
 
     last_ptr = NULL;
-    token = buffer;
-    for (token = strtok_r(token, ":", &last_ptr); token;
-         token = NULL, token = strtok_r(token, ":", &last_ptr)) {
+    for (token = strtok_r(buffer, ":", &last_ptr); token; token = strtok_r(NULL, ":", &last_ptr)) {
       if (!strcmp(token, "memory")) {
-        cgroup_ptr = strtok_r(NULL, ":", &last_ptr);
+        cgroup_ptr = strtok_r(NULL, "\n", &last_ptr);
```

Comment:
`strtok_r` is used here instead of `fgets` you mentioned.

containerd will use `:` as file name separator if cgroupfs is used so that it will read a wrong pod id and container id if search to `:`.